### PR TITLE
feat: Support for nested timestamp fields in Spark Offline store

### PR DESCRIPTION
**Pull request by EXPEbdodla**
# What this PR does / why we need it:
1. Supports using nested timestamp fields in the Spark Source using Field Mappings
2. Added Tests related to #4715 

# Which issue(s) this PR fixes:

# Misc

### CODE REVIEW

File: sdk/python/feast/utils.py [hunk 1]
```diff
@@ -212,6 +213,28 @@ def _run_pyarrow_field_mapping(
     return table
 
 
+def _get_fields_with_aliases(
```
Comment (franciscojavierarceo) [inline sdk/python/feast/utils.py:216]: probably would be good to add a test for this but i won't block on it.

**Review by franciscojavierarceo — APPROVED** [review]
lgtm, thank you for this!

File: sdk/python/feast/utils.py (hunk 1 above)
Comment (EXPEbdodla) [reply sdk/python/feast/utils.py:216]: Don't have a direct test but pull_latest_query tests will use it. I'll add explicit tests in next PRs.

**Comment by EXPEbdodla** [comment]
Resolved merge conflicts with another commit.

**Pull request merged**